minitask3/Lewis/follow.py

```diff
@@ -0,0 +1,334 @@
+from sensor_msgs.msg import Image, LaserScan
+from geometry_msgs.msg import Twist, PoseStamped
+from nav_msgs.msg import Path, Odometry
+from cv2 import cv2
+from cv_bridge import CvBridge, CvBridgeError
+from tf import transformations
+
+FORWARD_SPEED = 0.2
+TURN_SPEED = ((2 * PI) / 360) * 25  # 25 Deg/s
+WALK_DISTANCE = 3.0
+MIN_DISTANCE = 0.3
+
+class Follower():  # pylint: disable=too-many-instance-attributes
+    '''
+    Will beacon a Turtlebot3 Waffle toward green objects or wander around if not.
+    '''
+
+    def __init__(self):
+        rospy.on_shutdown(self.shutdown)
+        self.bridge = CvBridge()
+        self.image_sub = rospy.Subscriber('/camera/rgb/image_raw', Image, self.image_callback)
+        self.twist = Twist()
+        self.vel_pub = rospy.Publisher("/cmd_vel", Twist, queue_size=1)
+
+        # For path plotting
+        self.path_publisher = rospy.Publisher("/path", Path, queue_size=10)
+        self.path = Path()
+        self.plot_subscriber = rospy.Subscriber("odom", Odometry, self.plot_trajectory)
+
+        # For obstacle avoidance
+        self.laser_sub = rospy.Subscriber('/scan', LaserScan, self.scan_callback)
+        self.obstacle = True
+        self.direction = 0
+        self.direction_set = False
+
+        # For odometry and position tracking
+        self.curr_pose = {}
+        self.curr_pose["theta"] = 0.0
+        self.curr_pose["x"] = 0.0
+        self.curr_pose["y"] = 0.0
+        self.last_pose = {}
+        self.last_pose["theta"] = 0.0
+        self.last_pose["x"] = 0.0
+        self.last_pose["y"] = 0.0
+        self.turning = False
+        self.travel_angle = 0.0
+        self.object_detected = False
+        self.objects = 0
+        rospy.loginfo("No object detected")
+
+    def drive_turn_left(self):
+        '''
+        Turns left
+        '''
+        turn_left = Twist()
+        turn_left.linear.x = 0
+        turn_left.angular.z = TURN_SPEED
+        self.vel_pub.publish(turn_left)
+
+    def drive_turn_right(self):
+        '''
+        Turns right
+        '''
+        turn_right = Twist()
+        turn_right.linear.x = 0
+        turn_right.angular.z = -TURN_SPEED
+        self.vel_pub.publish(turn_right)
+
+    def drive_forward(self):
+        '''
+        Drives forwards
+        '''
+        forward = Twist()
+        forward.linear.x = FORWARD_SPEED
+        forward.angular.z = 0
+        self.vel_pub.publish(forward)
+
+    def drive_stop(self):
+        '''
+        Stops driving
+        '''
+        stop = Twist()
+        stop.linear.x = 0
+        stop.angular.z = -TURN_SPEED
+        self.vel_pub.publish(stop)
+
+    def image_callback(self, data):
+        '''
+        Handles the camera data.
+        '''
+        try:
+            cv_image = self.bridge.imgmsg_to_cv2(data,'bgr8')
+        except CvBridgeError as error:  # catch conversion errors
+            print(error)
+
+        cv_image_resized = cv2.resize(cv_image, (cv_image.shape[1]/4,cv_image.shape[0]/4))
```

Comment:
This line fill fail if line `102` fails and the except is triggered, as cv_image will be `None`

Reply:
You're also missing spaces around the `cv_image.shape[1]/4`, I would probably split that tuple into two lines.